### PR TITLE
Remove SwiftShader Docker images and software Vulkan testing.

### LINKED ISSUE #15299: Consider dropping SwiftShader as a software Vulkan test implementation
We've been using [SwiftShader](https://swiftshader.googlesource.com/SwiftShader)'s high-performance CPU-based implementation of Vulkan 1.3 for testing and development. Occasionally this has been a source of flaky tests (https://github.com/openxla/iree/issues/5716), but overall it has enabled us to run more tests on more platforms.

However, the SwiftShader project looks like it is on life support, and I worry that when we have a large Vulkan update or other testing environment changes occur, keeping it working may become more and more difficult. We are already seeing effects of this in Google's monorepo.

image

I'm not actually sure how much concrete value we're getting out of SwiftShader today

* We have `test_gpu` and `test_a100` CI jobs that run our Vulkan tests on physical desktop GPUs (Linux)
* We have a `build_and_test_android` CI job that runs our Vulkan tests on physical  mobile GPUs (Pixel 6 Pro, moto edge x30)
* Our Dockerfiles have a bunch of plumbing and other infrastructure for SwiftShader that we could prune
* We have CUDA, Metal, and ROCm backends, none of which have software implementations like Vulkan/SwiftShader
* SwiftShader does act as a "min spec" implementation, which is a good way to keep our implementation from growing too many advanced features as default-enabled

### PR DESCRIPTION
**Pull request by ScottTodd**
Fixes https://github.com/openxla/iree/issues/15299. This removes SwiftShader-flavored Dockerfiles from this repository and switches all uses across the project to use the non-SwiftShader equivalent images.

SwiftShader is a CPU implementation of the Vulkan API which we have been using for min-spec coverage and testing on devices that lack a physical GPU. We have CI workflows on most platforms we support that use real hardware and those builds are reliable enough to drop the CPU/SwiftShader coverage.

Vulkan tests are still included on presubmit using NVIDIA Tesla T4 GPUs in the `test_gpu` and `test_tf_integrations_gpu` jobs. Postsubmit jobs using NVIDIA A100 GPUs and Android phones (pixel-6-pro and moto-edge-x30) also run Vulkan tests.

Removing these extra Dockerfiles will also help us with the planned refactoring tracked in https://github.com/openxla/iree/issues/15332.

---

Follow-up work _not_ completed as part of this PR:

- [ ] Add min-spec coverage via profiles

### CODE REVIEW

File: build_tools/docker/prod_digests.txt [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
-gcr.io/iree-oss/emscripten@sha256:6e412f7ca51439ffce051a8a2e8fcbd7398743f461930a5f51c089441ffc3588
-gcr.io/iree-oss/android@sha256:d349a3d137d0ed0f7360af16096db682da0233679cdbb00a326150149e3c4fa4
+gcr.io/iree-oss/base@sha256:61e9aae211007dbad95e1f429e9e5121fd5968c204791038424979c21146cf75
+gcr.io/iree-oss/samples@sha256:86840f79a2d2c1896aaa4da9b2a3b33d822ba5bb24c8a5686cea9d6d07cddad3
+gcr.io/iree-oss/frontends@sha256:8e40a290cfd4c982645054b157691510a7aae5a0d5b53088ac491709a4d83f60
+gcr.io/iree-oss/frontends-nvidia@sha256:693afd91711e50e52a669eee074ad9007bf36745f4b1d1f5c3a19b82bd2d7c3a
+gcr.io/iree-oss/riscv@sha256:c630a314daf57b6920cb0ebae8aee2be1adab8e5180749f8460af7fb3ae83b66
+gcr.io/iree-oss/nvidia@sha256:892fefbdf90c93b407303adadfa87f22c0f1e84b7e819e69643c78fc5927c2ba
+gcr.io/iree-oss/emscripten@sha256:fe9bcd0851243c7b04d577d9e8a663860720484ec932bb53d4c56aec7f7b6d64
+gcr.io/iree-oss/android@sha256:66bc80fc16e31de0bf64cfcdc65be37a35b7f6c5cd44e89707e35d34da961bb5
```
Comment (ScottTodd) [inline build_tools/docker/prod_digests.txt:8]: FYI, here are the image sizes for these (as reported by `docker image inspect`):

Image name | Size (bytes)
-- | --
base | 6,526,634,381
samples | 8,840,964,409
frontends | 10,444,620,703
frontends-nvidia | 11,142,660,510
riscv | 10,541,452,432
nvidia | 21,315,311,794
emscripten | 7,645,068,636
android | 8,229,749,823

**Comment by benvanik** [comment]
ooc, any idea if we caught any spir-v codegen issues with swiftshader that weren't swiftshader bugs in the last year? I remember it caught some early on but don't recall anything recently

**Comment by ScottTodd** [comment]
> ooc, any idea if we caught any spir-v codegen issues with swiftshader that weren't swiftshader bugs in the last year? I remember it caught some early on but don't recall anything recently

I'm not remembering any cases of that. Maybe @antiagainst or @kuhar can recall?

**Comment by kuhar** [comment]
I remember one or two validation errors caught by swiftshader but IIRC these were caught locally by me, not in the CI.

**Comment by ScottTodd** [comment]
Ah, I found some validation layer errors on SwiftShader too, at https://github.com/openxla/iree/pull/15884#discussion_r1427375379, coming from SwiftShader using different device limits (e.g. 256 as `minStorageBufferOffsetAlignment`).

**Review by antiagainst — APPROVED** [review]

**Review by kuhar — APPROVED** [review]

**Pull request merged**